src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const allActions = []; // final array
+  const newState = { ...state };
+
+  for (const keys in actions) {
+    if (actions[keys].type === 'clear') {
+      for (const property in newState) {
+        delete newState[property];
+      }
+    }
+
+    if (actions[keys].type === 'addProperties') {
+      Object.assign(newState, actions[keys].extraData);
+    }
+
+    if (actions[keys].type === 'removeProperties') {
+      for (const action in actions[keys].keysToRemove) {
+        delete newState[actions[keys].keysToRemove[action]];
```

Comment:
split this single-line action into multi-line actions so that this code construction becomes more readable